From 393b4abe64e38b2e566fd7e438a2b6283a2c76d2 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Wed, 28 Jul 2021 20:11:35 +0300
Subject: [PATCH] [SW-45031] gaudi2: avoid pldm specific duplication

In this patch, Gaudi2 pldm HL_TARGET specific
common code moved to common dir.

Change-Id: I827a2e37e2d9b8905c87d076f38e1a75e71aa829
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/arc/pldm.c b/arch/gaudi2/arc/pldm.c
index aa4dd16..a6987cc 100644
--- a/arch/gaudi2/arc/pldm.c
+++ b/arch/gaudi2/arc/pldm.c
@@ -6,76 +6,10 @@
  */
 
 
-#include <stdbool.h>
-
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <psoc_cpu_pll_ctrl_regs.h>
-#include <psoc_global_conf_regs.h>
-#include "common.h"
 
-#define SYS_CLOCK_FREQUENCY	155000ULL
-
-/*
- * With Goya2 v88 PLDM image when PCI pll div3
- * is bumped up tp 250 MHz actual clock goes
- * to 12.9 KHz.
- */
-#define UART_CLOCK_FREQUENCY	12900ULL
-
-/* Working BAUD rate on PLDM */
-#define PLDM_UART_BAUDRATE	300
-
-/*
- * This cfg specific to PLDM bumps up uart freq
- * to 250 MHz while keeping other div's on refclk
- */
-/*
- * TODO: Enable this later
-static struct hl_pll_cfg gaudi2_pll_cfg[] = {
-	{
-		.name = "CPU", .base = mmPSOC_CPU_PLL_CTRL_BASE,
-		.nr = 1, .nf = 60, .od = 2, .nb = 3,
-		.div_factor = { 0, 0, 4, 1 },
-		.div_sel = { HL_PLL_REF_CLK,
-			HL_PLL_REF_CLK,
-			HL_PLL_REF_CLK,
-			HL_PLL_DIV_PLL
-		},
-	}
-};
-*/
 
 u32 hl_skip_pcie_init(void)
 {
 	return 0;
 }
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_dev_pll_init(void)
-{
-	/* TODO: Enable this later */
-	/* Bump up UART frequency */
-	//hl_pll_init(gaudi2_pll_cfg, 1);
-}
-
-void hl_uart_init(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor = 2; /* Min Divider */
-
-	/* baud rate doesn't support odd number */
-	return (factor + 1) & 0xFFFE;
-}
-void preboot_init_pcie_phy(void) {}
diff --git a/arch/gaudi2/arm/pldm.c b/arch/gaudi2/arm/pldm.c
index dc27f54..0548343 100644
--- a/arch/gaudi2/arm/pldm.c
+++ b/arch/gaudi2/arm/pldm.c
@@ -5,42 +5,13 @@
  *
  */
 
-#include <stdbool.h>
 
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
-#include <psoc_global_conf_regs.h>
 
-#define SYS_CLOCK_FREQUENCY	155000ULL
-#define UART_CLOCK_FREQUENCY	11900ULL
-#define PLDM_UART_BAUDRATE	300
 
 u32 hl_skip_pcie_init(void)
 {
 	return 1;
 }
 
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
-void hl_uart_init(void)
-{
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor = 2; /* Min Divider */
-
-	/* baud rate doesn't support odd number */
-	return (factor + 1) & 0xFFFE;
-}
-
 void preboot_sync_btl(void) {}
-void preboot_init_pcie_phy(void) {}
-
diff --git a/arch/gaudi2/common/pldm.c b/arch/gaudi2/common/pldm.c
index f5f8caf..672da66 100644
--- a/arch/gaudi2/common/pldm.c
+++ b/arch/gaudi2/common/pldm.c
@@ -5,4 +5,61 @@
  *
  */
 
-/* TODO: Gaudi2 pldm(HL_TARGET) specific common functions to be moved here */
+#include <preboot.h>
+#include <psoc_global_conf_regs.h>
+#include <uart.h>
+
+
+#define SYS_CLOCK_FREQUENCY	155000ULL
+
+#define UART_CLOCK_FREQUENCY	11900ULL
+#define PLDM_UART_BAUDRATE	300
+
+/*
+ * This cfg specific to PLDM bumps up uart freq
+ * to 250 MHz while keeping other div's on refclk
+ */
+/*
+ * TODO: Enable this later
+static struct hl_pll_cfg gaudi2_pll_cfg[] = {
+	{
+		.name = "CPU", .base = mmPSOC_CPU_PLL_CTRL_BASE,
+		.nr = 1, .nf = 60, .od = 2, .nb = 3,
+		.div_factor = { 0, 0, 4, 1 },
+		.div_sel = { HL_PLL_REF_CLK,
+			HL_PLL_REF_CLK,
+			HL_PLL_REF_CLK,
+			HL_PLL_DIV_PLL
+		},
+	}
+};
+*/
+
+u64 hl_pll_timestamp_freq_get(void)
+{
+	return SYS_CLOCK_FREQUENCY;
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor = 2; /* Min Divider */
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
+
+void hl_uart_init(void)
+{
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
+	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
+}
+
+void hl_dev_pll_init(void)
+{
+	/* TODO: Enable this later */
+	/* Bump up UART frequency */
+	//hl_pll_init(gaudi2_pll_cfg, 1);
+}
+
+void preboot_init_pcie_phy(void) {}
